tests/legacy-cli/e2e/tests/build/prod-build.ts

```diff
@@ -1,30 +1,37 @@
-import {join} from 'path';
-import {readdirSync} from 'fs';
-import {expectFileToExist, expectFileToMatch} from '../../utils/fs';
-import {ng} from '../../utils/process';
-import {expectGitToBeClean} from '../../utils/git';
+import { readdirSync } from 'fs';
+import { join } from 'path';
+import { getGlobalVariable } from '../../utils/env';
+import { expectFileToExist, expectFileToMatch } from '../../utils/fs';
+import { expectGitToBeClean } from '../../utils/git';
+import { ng } from '../../utils/process';
 
 
-export default function() {
+export default async function () {
   // TODO(architect): Delete this test. It is now in devkit/build-angular.
 
   // Can't use the `ng` helper because somewhere the environment gets
   // stuck to the first build done
-  return ng('build', '--prod')
-    .then(() => expectFileToExist(join(process.cwd(), 'dist')))
-    // Check for cache busting hash script src
-    .then(() => expectFileToMatch('dist/test-project/index.html', /main-es5\.[0-9a-f]{20}\.js/))
-    .then(() => expectFileToMatch('dist/test-project/index.html', /main-es2015\.[0-9a-f]{20}\.js/))
-    .then(() => expectFileToMatch('dist/test-project/index.html', /styles\.[0-9a-f]{20}\.css/))
-    .then(() => expectFileToMatch('dist/test-project/3rdpartylicenses.txt', /MIT/))
-    // Defaults to AoT
-    .then(() => {
-      const mainES5 = readdirSync('./dist/test-project').find(name => !!name.match(/main-es5.[a-z0-9]+\.js/));
-      expectFileToMatch(`dist/test-project/${mainES5}`, /bootstrapModuleFactory\(/);
+  const argv = getGlobalVariable('argv');
+  const ivyProject = argv['ivy'];
+  const bootstrapRegExp = ivyProject
+    ? /bootstrapModule\([a-z]+\)\./
+    : /bootstrapModuleFactory\([a-z]+\)\./;
 
-      const mainES2015 = readdirSync('./dist/test-project').find(name => !!name.match(/main-es2015.[a-z0-9]+\.js/));
-      expectFileToMatch(`dist/test-project/${mainES5}`, /bootstrapModuleFactory\(/);
```

Comment:
This regexp was actually matching the entire contents of the bundle, and in fact this test was not failing for Ivy